buster/documents_manager/service.py

```diff
@@ -53,14 +55,28 @@ def _add_documents(self, df: pd.DataFrame):
             source_id = self.get_source_id(source)
 
             df_source = df[df.source == source]
+            to_upsert = []
             for row in df_source.to_dict(orient="records"):
                 embedding = row["embedding"].tolist()
+                if use_sparse_vector:
+                    sparse_embedding = row["sparse_embedding"]
+
                 document = row.copy()
                 document.pop("embedding")
+                if use_sparse_vector:
+                    document.pop("sparse_embedding")
                 document["source_id"] = source_id
 
                 document_id = str(self.db.documents.insert_one(document).inserted_id)
-                self.index.upsert([(document_id, embedding, {"source": source})], namespace=self.namespace)
+                vector = {"id": document_id, "values": embedding, "metadata": {"source": source}}
+                if use_sparse_vector:
+                    vector["sparse_values"] = sparse_embedding
+
+                to_upsert.append(vector)
+
+            MAX_PINECONE_BATCH_SIZE = 100 if use_sparse_vector else 1000
```

Comment:
what is this batch size and why is it important it be different? maybe add a comment explaining the nuance?